packages/contracts-bridge/contracts/BridgeRouter.sol

```diff
@@ -240,38 +274,38 @@ contract BridgeRouter is Version0, Router {
      * @param _tokenId The token ID
      * @param _action The action
      * @param _fastEnabled True if fast liquidity was enabled, False otherwise
+     * @param _isConnext True if is connext transfer, False otherwise
      */
     function _handleTransfer(
         uint32 _origin,
         uint32 _nonce,
         bytes29 _tokenId,
         bytes29 _action,
-        bool _fastEnabled
+        bool _fastEnabled,
+        bool _isConnext
     ) internal {
         // get the token contract for the given tokenId on this chain;
         // (if the token is of remote origin and there is
         // no existing representation token contract, the TokenRegistry will
         // deploy a new one)
-        address _token = tokenRegistry.ensureLocalToken(
-            _tokenId.domain(),
-            _tokenId.id()
-        );
+        bytes32 _canonicalId = _tokenId.id();
+        uint32 _canonicalDomain = _tokenId.domain();
+        address _token = tokenRegistry.ensureLocalToken(_canonicalDomain, _canonicalId);
         // load the original recipient of the tokens
         address _recipient = _action.evmRecipient();
+        bytes32 _externalId;
         if (_fastEnabled) {
             // If an LP has prefilled this token transfer,
             // send the tokens to the LP instead of the recipient
-            bytes32 _id = BridgeMessage.getPreFillId(
-                _origin,
-                _nonce,
-                _tokenId,
-                _action
-            );
+            bytes32 _id = BridgeMessage.getPreFillId(_origin, _nonce, _tokenId, _action);
```

Comment:
we can deprecate this path in the next upgrade which will make this code less uggo